### ISSUE TITLE
Better ui design when hovering over words

### ISSUE THREAD

**Issue by mateo-meillon** [zed-industries/zed#7841]
### Check for existing issues

- [X] Completed

### Describe the feature

It would be nice if the popup and text would be one :))

### If applicable, add mockups / screenshots to help present your vision of the feature

Before:
<img width="677" alt="Screenshot 2024-02-15 at 16 26 08" src="https://github.com/zed-industries/zed/assets/59470689/cda6a94b-d035-4e52-8afb-3daf9607aa4b">

After:
<img width="677" alt="Screenshot 2024-02-15 at 16 26 08" src="https://github.com/zed-industries/zed/assets/59470689/1815451c-077a-482b-8229-ec4b6ab2e7c7">

**Comment by SouravInsights** [comment]
No one's currently working on this right? @JosephTLyons

**Comment by JosephTLyons** [comment]
> No one's currently working on this right? @JosephTLyons 

Nope, not to my knowledge. Would you like me to assign it to you?

**Comment by SouravInsights** [comment]
Would love to work on this but I haven't explored the codebase fully yet! It's my first time trying to contribute to Zed so I guess I'll have to spend some more time exploring the codebase first.. If it's not too much to ask can you please point me to the right crate I should be checking out for this issue?

**Comment by SomeoneToIgnore** [comment]
You're looking for https://github.com/zed-industries/zed/blob/e5d971f4c71aa2bb3c3aaea9d0838502c194992c/crates/editor/src/hover_popover.rs file, there are multiple flavors of those with their own `render` method responsible for drawing the thing.

**Comment by SouravInsights** [comment]
> You're looking for https://github.com/zed-industries/zed/blob/e5d971f4c71aa2bb3c3aaea9d0838502c194992c/crates/editor/src/hover_popover.rs file, there are multiple flavors of those with their own `render` method responsible for drawing the thing.

Yep found it out! But thanks anyway.. 🙂

**Comment by SouravInsights** [comment]
![CleanShot 2024-02-22 at 19 08 03@2x](https://github.com/zed-industries/zed/assets/26466468/4300e639-4d8e-4d32-8b83-f5e7e0327700)

What happens in a scenario like this? If the **`Popover Anchor`** & the **`Popover Element`** is combined together, do you think it will look odd? @JosephTLyons @mateo-meillon

**Comment by mateo-meillon** [comment]
@SouravInsights Haha, good question. Let me think about that

**Comment by mateo-meillon** [comment]
@SouravInsights  I would probably just make the top one artificially longer

**Comment by SouravInsights** [comment]
> @SouravInsights  I would probably just make the top one artificially longer

Could probably work!! How do we get some concrete feedback on this? @mateo-meillon @JosephTLyons

**Comment by JosephTLyons** [comment]
Good question - Going to lean on @SomeoneToIgnore for some advice here!

**Comment by SomeoneToIgnore** [comment]
VSCode:
<img width="653" alt="image" src="https://github.com/zed-industries/zed/assets/2690773/3c92af52-d22d-40ca-ac7d-6dc2715fb6d1">
<img width="658" alt="image" src="https://github.com/zed-industries/zed/assets/2690773/097be14c-38a4-49ba-83b8-f9cfcc9c2b16">

RustRover:
<img width="942" alt="image" src="https://github.com/zed-industries/zed/assets/2690773/f066dcce-3142-4035-952f-40e82e74e686">
<img width="665" alt="image" src="https://github.com/zed-industries/zed/assets/2690773/fe839f49-3b1b-4645-91e6-78d067b81d1e">


It looks like we can add some padding, customize the border a bit, and maybe add more contrast?
Not sure we need to get into anything fancy as that arrow-like edge of the bubble though.
